### ISSUE TITLE
Variables using Fn::If: does not resolve to a valid value

### ISSUE THREAD

**Issue by triqi** [arabold/serverless-export-env#6]
# Expected Behavior
Given an environment variable that uses a Fn::If: function
When I run `serverless export-env`
Then the output should result in a value that is evaluated from the function

# Current Behavior
The output resolves to [object Object]

# Steps to Reproduce
Declare a environment variable that uses a Fn::If: function
Define a condition in Resources to evaluate some expression
Deploy the stack
Run `serverless export-env`
Verify that the output `.env` file has exported the correct value

**Comment by arabold** [comment]
Interesting use case. Unfortunately I don't think there's a straight forward way of handling this case in the plugin.

The only solution I see without starting to reimplement a huge part of CloudFormation's logic (and going down the rabbit hole deeper and deeper) is by looking up the actual value from the deployed Lambda. But this has several drawbacks, i.e. you have to have the Lambda already deployed before `export-env` would be able to access the correct value. This can lead to weird and hard to debug errors if you're changing your logic in the `serverless.yaml` and want to test it without deploying first. So I'm not sure if that's a route we want to take at all.

The possibly simplest workaround would be to drop all `Fn::If:` from the environment variables and handle this logic in a `bootstrap.js` or similar. There you could set and unset `process.env` depending on your configuration needs.

**Comment by triqi** [comment]
@arabold you are right to suggest avoiding `Fn::If:` or any other delay-resolving CloudFormation intrinsic functions when using `serverless-export-env`. I will look into a post-deploy script to pull out the stack information I need.

Feel free to mark this issue as out-of-scope and close it.

Cheers.
